cpp/src/arrow/array/builder_binary.cc

```diff
@@ -162,7 +164,13 @@ void FixedSizeBinaryBuilder::Reset() {
 
 Status FixedSizeBinaryBuilder::Resize(int64_t capacity) {
   RETURN_NOT_OK(CheckCapacity(capacity));
-  RETURN_NOT_OK(byte_builder_.Resize(capacity * byte_width_));
+  int64_t dest_capacity_bytes;
```

Comment:
Do we still need this change?

Reply:
I don't know, I'll remove this and have a try

Reply:
I suggest we need this change, I add cout and run fuzzing and found there're so many error message now:

```
Status FixedSizeBinaryBuilder::Resize(int64_t capacity) {
  RETURN_NOT_OK(CheckCapacity(capacity));
  int64_t dest_capacity_bytes;
  if (ARROW_PREDICT_FALSE(
          MultiplyWithOverflow(capacity, byte_width_, &dest_capacity_bytes))) {
    std::cout << "Resize: capacity overflows (requested: " << capacity <<
                                 ", byte_width: " << byte_width_ << std::endl;
    return Status::CapacityError("Resize: capacity overflows (requested: ", capacity,
                                 ", byte_width: ", byte_width_, ")");
  }
  RETURN_NOT_OK(byte_builder_.Resize(dest_capacity_bytes));
  return ArrayBuilder::Resize(capacity);
}
```

Reply:
Oh, it's because `ReserveValues` is called above this:

```c++
  void ReserveValues(int64_t extra_values) override {
    ARROW_DCHECK(!uses_values_);
    TypedRecordReader::ReserveValues(extra_values);
    PARQUET_THROW_NOT_OK(array_builder_.Reserve(extra_values));
  }
```

`!uses_values_` is only called by ByteArray and FLBA, so other types doesn't touches this case...

Reply:
Yes, I think this change is ok. Sorry for the delay.